Framework/API/inc/MantidAPI/ExperimentInfo.h

```diff
@@ -120,6 +121,11 @@ class MANTID_API_DLL ExperimentInfo {
   void saveExperimentInfoNexus(::NeXus::File *file, bool saveLegacyInstrument = true) const;
   /// Saves this experiment description to the open NeXus file
   void saveExperimentInfoNexus(::NeXus::File *file, bool saveInstrument, bool saveSample, bool saveLogs) const;
+
+  void loadExperimentInfoNexus(const std::string &nxFilename, ::NeXus::File *file, std::string &parameterStr,
+                               const std::shared_ptr<Mantid::Kernel::NexusHDF5Descriptor> &fileInfo,
```

Comment:
Can we pass fileInfo as a reference (`Mantid::Kernel::NexusHDF5Descriptor &`) in this and similar functions?

Reply:
Good call.